dsp-user/handler/src/main/scala/dsp/user/handler/UserHandler.scala

```diff
@@ -23,107 +24,221 @@ import java.util.UUID
  * @param repo  the user repository
  */
 final case class UserHandler(repo: UserRepo) {
-  // implement all possible requests from V2, but divide things up into smaller functions to keep it cleaner than before
 
   /**
-   * Retrieve all users (sorted by IRI).
+   * Retrieves all users (sorted by IRI).
    */
   def getUsers(): UIO[List[User]] =
-    repo.getUsers().map(_.sorted)
-
-  // getSingleUserADM should be inspected in the route. According to the user identifier, the
-  // right method from the userHandler should be called.
+    repo.getUsers().map(_.sorted).tap(_ => ZIO.logInfo(s"Got all users"))
 
   /**
-   * Retrieve the user by ID.
+   * Retrieves the user by ID.
    *
    * @param id  the user's ID
    */
   def getUserById(id: UserId): IO[NotFoundException, User] =
     for {
-      user <- repo.getUserById(id).mapError(_ => NotFoundException("User not found"))
+      user <- repo
+                .getUserById(id)
+                .tap(_ => ZIO.logInfo(s"Looked up user by ID '${id}'"))
```

Comment:
I would add `.tap` at the end and after the `mapError`, as it is not part of the logic, but the `mapError` is.